MediaWikiKit/MediaWikiKit/MWKLocationSearchResult.h

```diff
@@ -2,10 +2,41 @@
 #import "MWKSearchResult.h"
 @import CoreLocation;
 
+/**
+ *  Response object model for search results which have geocoordinates.
+ *
+ *  @warning This object only supports deserialization <b>from</b> JSON, not serialization <b>to</b> JSON.
+ */
 @interface MWKLocationSearchResult : MWKSearchResult<MTLJSONSerializing>
 
+/**
+ *  Location serialized from the first set of coordinates in the response.
+ */
 @property (nonatomic, strong, readonly) CLLocation* location;
 
+/**
+ *  Number of meters between the receiver and the coordinate parameters of the originating search.
+ */
 @property (nonatomic, assign, readonly) CLLocationDistance distanceFromQueryCoordinates;
 
+/**
+ *  @name Automatically Updating Properties
+ *
+ *  These properties are set automatically by @c WMFNearbyViewModel in response to location & heading updates.
+ *
+ *  @see -[WMFNearbyViewModel autoUpdateResultAtIndex:]
+ */
+
+/**
+ *  Distance in meters from the user's current location.
+ */
+@property (nonatomic, assign) CLLocationDistance distanceFromUser;
```

Comment:
> What exactly do you think is confusing about the current properties? I thought distanceFromQueryCoordinates and distanceFromUser were pretty distinct.

Those properties are not part of the search result… they come from other places… which without length comments would not be clear whose responsibility it is to set them or when they may be triggered. 

I'm in favor of ditching purity for pragmatism… but I think its a little more than that here… we have properties that are calculated from the ether and then just hung on this object…  

If the creator of the object doesn't have the information to populate the object… that is a bit of a code smell to me.

> It also already has a reference to the search results, so IMHO it was intuitive to also manage updating their properties in response to events It was already responding to.

Yeah… but this wasn't required… the cell was already getting these updates without corrupting the 
model object. And its not clear what the "win" was in refactoring these properties be here instead.

If we are going to dump time into refactoring… we may as well do it right.